parquet-avro/src/main/java/org/apache/parquet/avro/AvroSchemaConverter.java

```diff
@@ -466,6 +466,21 @@ public Optional<Schema> visit(
                   LogicalTypeAnnotation.EnumLogicalTypeAnnotation enumLogicalType) {
                 return of(Schema.create(Schema.Type.STRING));
               }
+
+              @Override
+              public Optional<Schema> visit(
+                  LogicalTypeAnnotation.VariantLogicalTypeAnnotation variantLogicalType) {
+                String name = parquetGroupType.getName();
+                SchemaBuilder.FieldAssembler<Schema> builder = SchemaBuilder.builder(
+                        namespace(name, names))
+                    .record(name)
+                    .fields();
+                builder.name("metadata")
+                    .type(Schema.create(Schema.Type.BYTES))
+                    .noDefault();
+                builder.name("value").type().optional().type(Schema.create(Schema.Type.BYTES));
+                return of(builder.endRecord());
```

Comment:
I think we should also add a `variant` logical type name to the record.

Reply:
Yeah, I think that adding the logical type here would allow the caller to convert this record into a `Variant` representation using Avro logical type converters. Not a high priority, but it would be a good addition.